Author: Alastair McKinstry <mckinstry@debian.org>
Desctiption: DESC_INT needs to be declared before use, for gfortran to compile
 mmul_dir.h: gcc 'cpp' cant handle "recursive" PREFIX definitions within a header
  so preprocess.
Forwarded: no
Last-Updated: 2018-11-29

Index: flang-20180921/runtime/flang/mmul_cplx16contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_cplx16contmxm.F95
+++ flang-20180921/runtime/flang/mmul_cplx16contmxm.F95
@@ -18,13 +18,13 @@
 #include "mmul_dir.h"
 
 subroutine F90_matmul_cplx16_contmxm(dest, src1, src2, k_extent,m_extent,n_extent)
+ DESC_INT k_extent
+  DESC_INT n_extent
+  DESC_INT m_extent
 
   COMPLEX*16, dimension(k_extent,m_extent) :: src1
   COMPLEX*16, dimension(m_extent,n_extent) :: src2
   COMPLEX*16, dimension(k_extent,n_extent) :: dest
-  DESC_INT k_extent
-  DESC_INT n_extent
-  DESC_INT m_extent
 
   DESC_INT k
   DESC_INT n
Index: flang-20180921/runtime/flang/mmul_cplx8contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_cplx8contmxm.F95
+++ flang-20180921/runtime/flang/mmul_cplx8contmxm.F95
@@ -19,12 +19,12 @@
 
 subroutine F90_matmul_cplx8_contmxm(dest, src1, src2, k_extent, m_extent, n_extent)
 
-  COMPLEX*8, dimension(k_extent,m_extent) :: src1
-  COMPLEX*8, dimension(m_extent,n_extent) :: src2
-  COMPLEX*8, dimension(k_extent,n_extent) :: dest
   DESC_INT k_extent
   DESC_INT n_extent
   DESC_INT m_extent
+  COMPLEX*8, dimension(k_extent,m_extent) :: src1
+  COMPLEX*8, dimension(m_extent,n_extent) :: src2
+  COMPLEX*8, dimension(k_extent,n_extent) :: dest
 
   DESC_INT k
   DESC_INT n
Index: flang-20180921/runtime/flang/mmul_int1contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_int1contmxm.F95
+++ flang-20180921/runtime/flang/mmul_int1contmxm.F95
@@ -18,13 +18,13 @@
 #include "mmul_dir.h"
 
 subroutine F90_matmul_int1_contmxm(dest, src1, src2, k_extent, m_extent, n_extent)
+  DESC_INT k_extent
+  DESC_INT n_extent
+  DESC_INT m_extent
 
   INTEGER*1, dimension(k_extent,m_extent) :: src1
   INTEGER*1, dimension(m_extent,n_extent) :: src2
   INTEGER*1, dimension(k_extent,n_extent) :: dest
-  DESC_INT k_extent
-  DESC_INT n_extent
-  DESC_INT m_extent
 
   DESC_INT k
   DESC_INT n
Index: flang-20180921/runtime/flang/mmul_int2contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_int2contmxm.F95
+++ flang-20180921/runtime/flang/mmul_int2contmxm.F95
@@ -19,12 +19,12 @@
 
 subroutine F90_matmul_int2_contmxm(dest, src1, src2, k_extent, m_extent, n_extent)
 
-  INTEGER*2, dimension(k_extent,m_extent):: src1
-  INTEGER*2, dimension(m_extent,n_extent) :: src2
-  INTEGER*2, dimension(k_extent,n_extent) :: dest
   DESC_INT k_extent
   DESC_INT n_extent
   DESC_INT m_extent
+  INTEGER*2, dimension(k_extent,m_extent):: src1
+  INTEGER*2, dimension(m_extent,n_extent) :: src2
+  INTEGER*2, dimension(k_extent,n_extent) :: dest
 
   DESC_INT k
   DESC_INT n
Index: flang-20180921/runtime/flang/mmul_int4contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_int4contmxm.F95
+++ flang-20180921/runtime/flang/mmul_int4contmxm.F95
@@ -19,12 +19,12 @@
 
 subroutine F90_matmul_int4_contmxm(dest, src1, src2, k_extent, m_extent, n_extent)
 
-  INTEGER*4, dimension(k_extent,m_extent) :: src1
-  INTEGER*4, dimension(m_extent,n_extent) :: src2
-  INTEGER*4, dimension(k_extent,n_extent) :: dest
   DESC_INT k_extent
   DESC_INT n_extent
   DESC_INT m_extent
+  INTEGER*4, dimension(k_extent,m_extent) :: src1
+  INTEGER*4, dimension(m_extent,n_extent) :: src2
+  INTEGER*4, dimension(k_extent,n_extent) :: dest
 
   DESC_INT k
   DESC_INT n
Index: flang-20180921/runtime/flang/mmul_int8contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_int8contmxm.F95
+++ flang-20180921/runtime/flang/mmul_int8contmxm.F95
@@ -19,12 +19,12 @@
 
 subroutine F90_matmul_int8_contmxm(dest, src1, src2, k_extent, m_extent, n_extent)
 
-  INTEGER*8, dimension(k_extent,m_extent) :: src1
-  INTEGER*8, dimension(m_extent,n_extent) :: src2
-  INTEGER*8, dimension(k_extent,n_extent) :: dest
   DESC_INT k_extent
   DESC_INT n_extent
   DESC_INT m_extent
+  INTEGER*8, dimension(k_extent,m_extent) :: src1
+  INTEGER*8, dimension(m_extent,n_extent) :: src2
+  INTEGER*8, dimension(k_extent,n_extent) :: dest
 
   DESC_INT k
   DESC_INT n
Index: flang-20180921/runtime/flang/mmul_log1contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_log1contmxm.F95
+++ flang-20180921/runtime/flang/mmul_log1contmxm.F95
@@ -19,12 +19,12 @@
 
 subroutine F90_matmul_log1_contmxm(dest, src1, src2, k_extent, m_extent, n_extent)
 
-  LOGICAL*1, dimension(k_extent,m_extent) :: src1
-  LOGICAL*1, dimension(m_extent,n_extent) :: src2
-  LOGICAL*1, dimension(k_extent,n_extent) :: dest
   DESC_INT k_extent
   DESC_INT n_extent
   DESC_INT m_extent
+  LOGICAL*1, dimension(k_extent,m_extent) :: src1
+  LOGICAL*1, dimension(m_extent,n_extent) :: src2
+  LOGICAL*1, dimension(k_extent,n_extent) :: dest
 
   DESC_INT k
   DESC_INT n
Index: flang-20180921/runtime/flang/mmul_log2contmxm.F95
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_log2contmxm.F95
+++ flang-20180921/runtime/flang/mmul_log2contmxm.F95
@@ -19,12 +19,12 @@
 
 subroutine F90_matmul_log2_contmxm(dest, src1, src2, k_extent, m_extent, n_extent)
 
-  LOGICAL*2, dimension(k_extent,m_extent) :: src1
-  LOGICAL*2, dimension(m_extent,n_extent) :: src2
-  LOGICAL*2, dimension(k_extent,n_extent) :: dest
   DESC_INT k_extent
   DESC_INT n_extent
   DESC_INT m_extent
+  LOGICAL*2, dimension(k_extent,m_extent) :: src1
+  LOGICAL*2, dimension(m_extent,n_extent) :: src2
+  LOGICAL*2, dimension(k_extent,n_extent) :: dest
 
   DESC_INT k
   DESC_INT n
@@ -32,7 +32,7 @@ subroutine F90_matmul_log2_contmxm(dest,
 
   do n=1,n_extent
     do k=1,k_extent
-      dest(k,n) = 0
+      dest(k,n) = .FALSE.
     end do
   end do
   do n=1,n_extent
Index: flang-20180921/runtime/flang/mmul_dir.h
===================================================================
--- flang-20180921.orig/runtime/flang/mmul_dir.h
+++ flang-20180921/runtime/flang/mmul_dir.h
@@ -1,7 +1,7 @@
 ! 
-! Copyright (c) 2004-2018, NVIDIA CORPORATION.  All rights reserved.
+! Copyright (c 2004-2018, NVIDIA CORPORATION.  All rights reserved.
 !
-! Licensed under the Apache License, Version 2.0 (the "License");
+! Licensed under the Apache License, Version 2.0 (the "License";
 ! you may not use this file except in compliance with the License.
 ! You may obtain a copy of the License at
 !
@@ -17,165 +17,163 @@
 
 ! directives.h -- contains preprocessor directives for F90 rte files
 
-#define PREFIX(x) f90##x
-#define _PREFIX(x) _f90##x
 
 #ifdef DESC_I8
 #define DESC_INT INTEGER(8)
-#define F90_matmul_cplx16_contmxm	PREFIX(_mm_cplx16_contmxm_i8)
-#define F90_matmul_cplx16_contmxv	PREFIX(_mm_cplx16_contmxv_i8)
-#define F90_matmul_cplx16_contvxm	PREFIX(_mm_cplx16_contvxm_i8)
-#define F90_matmul_cplx16_str1		PREFIX(_mm_cplx16_str1_i8)
-#define F90_matmul_cplx16_str1_mxv	PREFIX(_mm_cplx16_str1_mxv_i8)
-#define F90_matmul_cplx16_str1_mxv_t	PREFIX(_mm_cplx16_str1_mxv_t_i8)
-#define F90_matmul_cplx16_str1_vxm	PREFIX(_mm_cplx16_str1_vxm_i8)
-#define F90_matmul_cplx32_contmxm	PREFIX(_mm_cplx32_contmxm_i8)
-#define F90_matmul_cplx32_contmxv	PREFIX(_mm_cplx32_contmxv_i8)
-#define F90_matmul_cplx32_contvxm	PREFIX(_mm_cplx32_contvxm_i8)
-#define F90_matmul_cplx32_str1		PREFIX(_mm_cplx32_str1_i8)
-#define F90_matmul_cplx32_str1_mxv	PREFIX(_mm_cplx32_str1_mxv_i8)
-#define F90_matmul_cplx8_contmxm	PREFIX(_mm_cplx8_contmxm_i8)
-#define F90_matmul_cplx8_contmxv	PREFIX(_mm_cplx8_contmxv_i8)
-#define F90_matmul_cplx8_contvxm	PREFIX(_mm_cplx8_contvxm_i8)
-#define F90_matmul_cplx8_str1		PREFIX(_mm_cplx8_str1_i8)
-#define F90_matmul_cplx8_str1_mxv	PREFIX(_mm_cplx8_str1_mxv_i8)
-#define F90_matmul_cplx8_str1_mxv_t	PREFIX(_mm_cplx8_str1_mxv_t_i8)
-#define F90_matmul_cplx8_str1_t		PREFIX(_mm_cplx8_str1_t_i8)
-#define F90_matmul_cplx8_str1_vxm	PREFIX(_mm_cplx8_str1_vxm_i8)
-#define F90_matmul_int1_contmxm		PREFIX(_mm_int1_contmxm_i8)
-#define F90_matmul_int1_contmxv		PREFIX(_mm_int1_contmxv_i8)
-#define F90_matmul_int1_contvxm		PREFIX(_mm_int1_contvxm_i8)
-#define F90_matmul_int1_str1		PREFIX(_mm_int1_str1_i8)
-#define F90_matmul_int1_str1_mxv	PREFIX(_mm_int1_str1_mxv_i8)
-#define F90_matmul_int1_str1_vxm	PREFIX(_mm_int1_str1_vxm_i8)
-#define F90_matmul_int2_contmxm		PREFIX(_mm_int2_contmxm_i8)
-#define F90_matmul_int2_contmxv		PREFIX(_mm_int2_contmxv_i8)
-#define F90_matmul_int2_contvxm		PREFIX(_mm_int2_contvxm_i8)
-#define F90_matmul_int2_str1		PREFIX(_mm_int2_str1_i8)
-#define F90_matmul_int2_str1_mxv	PREFIX(_mm_int2_str1_mxv_i8)
-#define F90_matmul_int2_str1_vxm	PREFIX(_mm_int2_str1_vxm_i8)
-#define F90_matmul_int4_contmxm		PREFIX(_mm_int4_contmxm_i8)
-#define F90_matmul_int4_contmxv		PREFIX(_mm_int4_contmxv_i8)
-#define F90_matmul_int4_contvxm		PREFIX(_mm_int4_contvxm_i8)
-#define F90_matmul_int4_str1		PREFIX(_mm_int4_str1_i8)
-#define F90_matmul_int4_str1_mxv	PREFIX(_mm_int4_str1_mxv_i8)
-#define F90_matmul_int4_str1_vxm	PREFIX(_mm_int4_str1_vxm_i8)
-#define F90_matmul_int8_contmxm		PREFIX(_mm_int8_contmxm_i8)
-#define F90_matmul_int8_contmxv		PREFIX(_mm_int8_contmxv_i8)
-#define F90_matmul_int8_contvxm		PREFIX(_mm_int8_contvxm_i8)
-#define F90_matmul_int8_str1		PREFIX(_mm_int8_str1_i8)
-#define F90_matmul_int8_str1_mxv	PREFIX(_mm_int8_str1_mxv_i8)
-#define F90_matmul_int8_str1_vxm	PREFIX(_mm_int8_str1_vxm_i8)
-#define F90_matmul_log1_contmxm		PREFIX(_mm_log1_contmxm_i8)
-#define F90_matmul_log1_contmxv		PREFIX(_mm_log1_contmxv_i8)
-#define F90_matmul_log1_contvxm		PREFIX(_mm_log1_contvxm_i8)
-#define F90_matmul_log2_contmxm		PREFIX(_mm_log2_contmxm_i8)
-#define F90_matmul_log2_contmxv		PREFIX(_mm_log2_contmxv_i8)
-#define F90_matmul_log2_contvxm		PREFIX(_mm_log2_contvxm_i8)
-#define F90_matmul_log4_contmxm		PREFIX(_mm_log4_contmxm_i8)
-#define F90_matmul_log4_contmxv		PREFIX(_mm_log4_contmxv_i8)
-#define F90_matmul_log4_contvxm		PREFIX(_mm_log4_contvxm_i8)
-#define F90_matmul_log8_contmxm		PREFIX(_mm_log8_contmxm_i8)
-#define F90_matmul_log8_contmxv		PREFIX(_mm_log8_contmxv_i8)
-#define F90_matmul_log8_contvxm		PREFIX(_mm_log8_contvxm_i8)
-#define F90_matmul_real16_contmxm	PREFIX(_mm_real16_contmxm_i8)
-#define F90_matmul_real16_contmxv	PREFIX(_mm_real16_contmxv_i8)
-#define F90_matmul_real16_contvxm	PREFIX(_mm_real16_contvxm_i8)
-#define F90_matmul_real4_contmxm	PREFIX(_mm_real4_contmxm_i8)
-#define F90_matmul_real4_contmxv	PREFIX(_mm_real4_contmxv_i8)
-#define F90_matmul_real4_contvxm	PREFIX(_mm_real4_contvxm_i8)
-#define F90_matmul_real4_str1		PREFIX(_mm_real4_str1_i8)
-#define F90_matmul_real4_str1_mxv	PREFIX(_mm_real4_str1_mxv_i8)
-#define F90_matmul_real4_str1_mxv_t	PREFIX(_mm_real4_str1_mxv_t_i8)
-#define F90_matmul_real4_str1_t		PREFIX(_mm_real4_str1_t_i8)
-#define F90_matmul_real4_str1_vxm	PREFIX(_mm_real4_str1_vxm_i8)
-#define F90_matmul_real8_contmxm	PREFIX(_mm_real8_contmxm_i8)
-#define F90_matmul_real8_contmxv	PREFIX(_mm_real8_contmxv_i8)
-#define F90_matmul_real8_contvxm	PREFIX(_mm_real8_contvxm_i8)
-#define F90_matmul_real8_str1		PREFIX(_mm_real8_str1_i8)
-#define _F90_matmul_real8_str1a		_PREFIX(_mm_real8_str1a_i8)
-#define _F90_matmul_real8_str1b		_PREFIX(_mm_real8_str1b_i8)
-#define _F90_matmul_real8_str1c		_PREFIX(_mm_real8_str1c_i8)
-#define F90_matmul_real8_str1_mxv	PREFIX(_mm_real8_str1_mxv_i8)
-#define F90_matmul_real8_str1_mxv_t	PREFIX(_mm_real8_str1_mxv_t_i8)
-#define F90_matmul_real8_str1_vxm	PREFIX(_mm_real8_str1_vxm_i8)
+#define F90_matmul_cplx16_contmxm	f90_mm_cplx16_contmxm_i8
+#define F90_matmul_cplx16_contmxv	f90_mm_cplx16_contmxv_i8
+#define F90_matmul_cplx16_contvxm	f90_mm_cplx16_contvxm_i8
+#define F90_matmul_cplx16_str1		f90_mm_cplx16_str1_i8
+#define F90_matmul_cplx16_str1_mxv	f90_mm_cplx16_str1_mxv_i8
+#define F90_matmul_cplx16_str1_mxv_t	f90_mm_cplx16_str1_mxv_t_i8
+#define F90_matmul_cplx16_str1_vxm	f90_mm_cplx16_str1_vxm_i8
+#define F90_matmul_cplx32_contmxm	f90_mm_cplx32_contmxm_i8
+#define F90_matmul_cplx32_contmxv	f90_mm_cplx32_contmxv_i8
+#define F90_matmul_cplx32_contvxm	f90_mm_cplx32_contvxm_i8
+#define F90_matmul_cplx32_str1		f90_mm_cplx32_str1_i8
+#define F90_matmul_cplx32_str1_mxv	f90_mm_cplx32_str1_mxv_i8
+#define F90_matmul_cplx8_contmxm	f90_mm_cplx8_contmxm_i8
+#define F90_matmul_cplx8_contmxv	f90_mm_cplx8_contmxv_i8
+#define F90_matmul_cplx8_contvxm	f90_mm_cplx8_contvxm_i8
+#define F90_matmul_cplx8_str1		f90_mm_cplx8_str1_i8
+#define F90_matmul_cplx8_str1_mxv	f90_mm_cplx8_str1_mxv_i8
+#define F90_matmul_cplx8_str1_mxv_t	f90_mm_cplx8_str1_mxv_t_i8
+#define F90_matmul_cplx8_str1_t		f90_mm_cplx8_str1_t_i8
+#define F90_matmul_cplx8_str1_vxm	f90_mm_cplx8_str1_vxm_i8
+#define F90_matmul_int1_contmxm		f90_mm_int1_contmxm_i8
+#define F90_matmul_int1_contmxv		f90_mm_int1_contmxv_i8
+#define F90_matmul_int1_contvxm		f90_mm_int1_contvxm_i8
+#define F90_matmul_int1_str1		f90_mm_int1_str1_i8
+#define F90_matmul_int1_str1_mxv	f90_mm_int1_str1_mxv_i8
+#define F90_matmul_int1_str1_vxm	f90_mm_int1_str1_vxm_i8
+#define F90_matmul_int2_contmxm		f90_mm_int2_contmxm_i8
+#define F90_matmul_int2_contmxv		f90_mm_int2_contmxv_i8
+#define F90_matmul_int2_contvxm		f90_mm_int2_contvxm_i8
+#define F90_matmul_int2_str1		f90_mm_int2_str1_i8
+#define F90_matmul_int2_str1_mxv	f90_mm_int2_str1_mxv_i8
+#define F90_matmul_int2_str1_vxm	f90_mm_int2_str1_vxm_i8
+#define F90_matmul_int4_contmxm		f90_mm_int4_contmxm_i8
+#define F90_matmul_int4_contmxv		f90_mm_int4_contmxv_i8
+#define F90_matmul_int4_contvxm		f90_mm_int4_contvxm_i8
+#define F90_matmul_int4_str1		f90_mm_int4_str1_i8
+#define F90_matmul_int4_str1_mxv	f90_mm_int4_str1_mxv_i8
+#define F90_matmul_int4_str1_vxm	f90_mm_int4_str1_vxm_i8
+#define F90_matmul_int8_contmxm		f90_mm_int8_contmxm_i8
+#define F90_matmul_int8_contmxv		f90_mm_int8_contmxv_i8
+#define F90_matmul_int8_contvxm		f90_mm_int8_contvxm_i8
+#define F90_matmul_int8_str1		f90_mm_int8_str1_i8
+#define F90_matmul_int8_str1_mxv	f90_mm_int8_str1_mxv_i8
+#define F90_matmul_int8_str1_vxm	f90_mm_int8_str1_vxm_i8
+#define F90_matmul_log1_contmxm		f90_mm_log1_contmxm_i8
+#define F90_matmul_log1_contmxv		f90_mm_log1_contmxv_i8
+#define F90_matmul_log1_contvxm		f90_mm_log1_contvxm_i8
+#define F90_matmul_log2_contmxm		f90_mm_log2_contmxm_i8
+#define F90_matmul_log2_contmxv		f90_mm_log2_contmxv_i8
+#define F90_matmul_log2_contvxm		f90_mm_log2_contvxm_i8
+#define F90_matmul_log4_contmxm		f90_mm_log4_contmxm_i8
+#define F90_matmul_log4_contmxv		f90_mm_log4_contmxv_i8
+#define F90_matmul_log4_contvxm		f90_mm_log4_contvxm_i8
+#define F90_matmul_log8_contmxm		f90_mm_log8_contmxm_i8
+#define F90_matmul_log8_contmxv		f90_mm_log8_contmxv_i8
+#define F90_matmul_log8_contvxm		f90_mm_log8_contvxm_i8
+#define F90_matmul_real16_contmxm	f90_mm_real16_contmxm_i8
+#define F90_matmul_real16_contmxv	f90_mm_real16_contmxv_i8
+#define F90_matmul_real16_contvxm	f90_mm_real16_contvxm_i8
+#define F90_matmul_real4_contmxm	f90_mm_real4_contmxm_i8
+#define F90_matmul_real4_contmxv	f90_mm_real4_contmxv_i8
+#define F90_matmul_real4_contvxm	f90_mm_real4_contvxm_i8
+#define F90_matmul_real4_str1		f90_mm_real4_str1_i8
+#define F90_matmul_real4_str1_mxv	f90_mm_real4_str1_mxv_i8
+#define F90_matmul_real4_str1_mxv_t	f90_mm_real4_str1_mxv_t_i8
+#define F90_matmul_real4_str1_t		f90_mm_real4_str1_t_i8
+#define F90_matmul_real4_str1_vxm	f90_mm_real4_str1_vxm_i8
+#define F90_matmul_real8_contmxm	f90_mm_real8_contmxm_i8
+#define F90_matmul_real8_contmxv	f90_mm_real8_contmxv_i8
+#define F90_matmul_real8_contvxm	f90_mm_real8_contvxm_i8
+#define F90_matmul_real8_str1		f90_mm_real8_str1_i8
+#define _F90_matmul_real8_str1a		_f90_mm_real8_str1a_i8
+#define _F90_matmul_real8_str1b		_f90_mm_real8_str1b_i8
+#define _F90_matmul_real8_str1c		_f90_mm_real8_str1c_i8
+#define F90_matmul_real8_str1_mxv	f90_mm_real8_str1_mxv_i8
+#define F90_matmul_real8_str1_mxv_t	f90_mm_real8_str1_mxv_t_i8
+#define F90_matmul_real8_str1_vxm	f90_mm_real8_str1_vxm_i8
 #else
 #define DESC_INT INTEGER(4)
-#define F90_matmul_cplx16_contmxm	PREFIX(_mm_cplx16_contmxm)
-#define F90_matmul_cplx16_contmxv	PREFIX(_mm_cplx16_contmxv)
-#define F90_matmul_cplx16_contvxm	PREFIX(_mm_cplx16_contvxm)
-#define F90_matmul_cplx16_str1		PREFIX(_mm_cplx16_str1)
-#define F90_matmul_cplx16_str1_mxv	PREFIX(_mm_cplx16_str1_mxv)
-#define F90_matmul_cplx16_str1_mxv_t	PREFIX(_mm_cplx16_str1_mxv_t)
-#define F90_matmul_cplx16_str1_vxm	PREFIX(_mm_cplx16_str1_vxm)
-#define F90_matmul_cplx32_contmxm	PREFIX(_mm_cplx32_contmxm)
-#define F90_matmul_cplx32_contmxv	PREFIX(_mm_cplx32_contmxv)
-#define F90_matmul_cplx32_contvxm	PREFIX(_mm_cplx32_contvxm)
-#define F90_matmul_cplx32_str1		PREFIX(_mm_cplx32_str1)
-#define F90_matmul_cplx32_str1_mxv	PREFIX(_mm_cplx32_str1_mxv)
-#define F90_matmul_cplx8_contmxm	PREFIX(_mm_cplx8_contmxm)
-#define F90_matmul_cplx8_contmxv	PREFIX(_mm_cplx8_contmxv)
-#define F90_matmul_cplx8_contvxm	PREFIX(_mm_cplx8_contvxm)
-#define F90_matmul_cplx8_str1		PREFIX(_mm_cplx8_str1)
-#define F90_matmul_cplx8_str1_mxv	PREFIX(_mm_cplx8_str1_mxv)
-#define F90_matmul_cplx8_str1_mxv_t	PREFIX(_mm_cplx8_str1_mxv_t)
-#define F90_matmul_cplx8_str1_t		PREFIX(_mm_cplx8_str1_t)
-#define F90_matmul_cplx8_str1_vxm	PREFIX(_mm_cplx8_str1_vxm)
-#define F90_matmul_int1_contmxm		PREFIX(_mm_int1_contmxm)
-#define F90_matmul_int1_contmxv		PREFIX(_mm_int1_contmxv)
-#define F90_matmul_int1_contvxm		PREFIX(_mm_int1_contvxm)
-#define F90_matmul_int1_str1		PREFIX(_mm_int1_str1)
-#define F90_matmul_int1_str1_mxv	PREFIX(_mm_int1_str1_mxv)
-#define F90_matmul_int1_str1_vxm	PREFIX(_mm_int1_str1_vxm)
-#define F90_matmul_int2_contmxm		PREFIX(_mm_int2_contmxm)
-#define F90_matmul_int2_contmxv		PREFIX(_mm_int2_contmxv)
-#define F90_matmul_int2_contvxm		PREFIX(_mm_int2_contvxm)
-#define F90_matmul_int2_str1		PREFIX(_mm_int2_str1)
-#define F90_matmul_int2_str1_mxv	PREFIX(_mm_int2_str1_mxv)
-#define F90_matmul_int2_str1_vxm	PREFIX(_mm_int2_str1_vxm)
-#define F90_matmul_int4_contmxm		PREFIX(_mm_int4_contmxm)
-#define F90_matmul_int4_contmxv		PREFIX(_mm_int4_contmxv)
-#define F90_matmul_int4_contvxm		PREFIX(_mm_int4_contvxm)
-#define F90_matmul_int4_str1		PREFIX(_mm_int4_str1)
-#define F90_matmul_int4_str1_mxv	PREFIX(_mm_int4_str1_mxv)
-#define F90_matmul_int4_str1_vxm	PREFIX(_mm_int4_str1_vxm)
-#define F90_matmul_int8_contmxm		PREFIX(_mm_int8_contmxm)
-#define F90_matmul_int8_contmxv		PREFIX(_mm_int8_contmxv)
-#define F90_matmul_int8_contvxm		PREFIX(_mm_int8_contvxm)
-#define F90_matmul_int8_str1		PREFIX(_mm_int8_str1)
-#define F90_matmul_int8_str1_mxv	PREFIX(_mm_int8_str1_mxv)
-#define F90_matmul_int8_str1_vxm	PREFIX(_mm_int8_str1_vxm)
-#define F90_matmul_log1_contmxm		PREFIX(_mm_log1_contmxm)
-#define F90_matmul_log1_contmxv		PREFIX(_mm_log1_contmxv)
-#define F90_matmul_log1_contvxm		PREFIX(_mm_log1_contvxm)
-#define F90_matmul_log2_contmxm		PREFIX(_mm_log2_contmxm)
-#define F90_matmul_log2_contmxv		PREFIX(_mm_log2_contmxv)
-#define F90_matmul_log2_contvxm		PREFIX(_mm_log2_contvxm)
-#define F90_matmul_log4_contmxm		PREFIX(_mm_log4_contmxm)
-#define F90_matmul_log4_contmxv		PREFIX(_mm_log4_contmxv)
-#define F90_matmul_log4_contvxm		PREFIX(_mm_log4_contvxm)
-#define F90_matmul_log8_contmxm		PREFIX(_mm_log8_contmxm)
-#define F90_matmul_log8_contmxv		PREFIX(_mm_log8_contmxv)
-#define F90_matmul_log8_contvxm		PREFIX(_mm_log8_contvxm)
-#define F90_matmul_real16_contmxm	PREFIX(_mm_real16_contmxm)
-#define F90_matmul_real16_contmxv	PREFIX(_mm_real16_contmxv)
-#define F90_matmul_real16_contvxm	PREFIX(_mm_real16_contvxm)
-#define F90_matmul_real4_contmxm	PREFIX(_mm_real4_contmxm)
-#define F90_matmul_real4_contmxv	PREFIX(_mm_real4_contmxv)
-#define F90_matmul_real4_contvxm	PREFIX(_mm_real4_contvxm)
-#define F90_matmul_real4_str1		PREFIX(_mm_real4_str1)
-#define F90_matmul_real4_str1_mxv	PREFIX(_mm_real4_str1_mxv)
-#define F90_matmul_real4_str1_mxv_t	PREFIX(_mm_real4_str1_mxv_t)
-#define F90_matmul_real4_str1_t		PREFIX(_mm_real4_str1_t)
-#define F90_matmul_real4_str1_vxm	PREFIX(_mm_real4_str1_vxm)
-#define F90_matmul_real8_contmxm	PREFIX(_mm_real8_contmxm)
-#define F90_matmul_real8_contmxv	PREFIX(_mm_real8_contmxv)
-#define F90_matmul_real8_contvxm	PREFIX(_mm_real8_contvxm)
-#define F90_matmul_real8_str1		PREFIX(_mm_real8_str1)
-#define _F90_matmul_real8_str1a		_PREFIX(_mm_real8_str1a)
-#define _F90_matmul_real8_str1b		_PREFIX(_mm_real8_str1b)
-#define _F90_matmul_real8_str1c		_PREFIX(_mm_real8_str1c)
-#define F90_matmul_real8_str1_mxv	PREFIX(_mm_real8_str1_mxv)
-#define F90_matmul_real8_str1_mxv_t	PREFIX(_mm_real8_str1_mxv_t)
-#define F90_matmul_real8_str1_vxm	PREFIX(_mm_real8_str1_vxm)
+#define F90_matmul_cplx16_contmxm	f90_mm_cplx16_contmxm
+#define F90_matmul_cplx16_contmxv	f90_mm_cplx16_contmxv
+#define F90_matmul_cplx16_contvxm	f90_mm_cplx16_contvxm
+#define F90_matmul_cplx16_str1		f90_mm_cplx16_str1
+#define F90_matmul_cplx16_str1_mxv	f90_mm_cplx16_str1_mxv
+#define F90_matmul_cplx16_str1_mxv_t	f90_mm_cplx16_str1_mxv_t
+#define F90_matmul_cplx16_str1_vxm	f90_mm_cplx16_str1_vxm
+#define F90_matmul_cplx32_contmxm	f90_mm_cplx32_contmxm
+#define F90_matmul_cplx32_contmxv	f90_mm_cplx32_contmxv
+#define F90_matmul_cplx32_contvxm	f90_mm_cplx32_contvxm
+#define F90_matmul_cplx32_str1		f90_mm_cplx32_str1
+#define F90_matmul_cplx32_str1_mxv	f90_mm_cplx32_str1_mxv
+#define F90_matmul_cplx8_contmxm	f90_mm_cplx8_contmxm
+#define F90_matmul_cplx8_contmxv	f90_mm_cplx8_contmxv
+#define F90_matmul_cplx8_contvxm	f90_mm_cplx8_contvxm
+#define F90_matmul_cplx8_str1		f90_mm_cplx8_str1
+#define F90_matmul_cplx8_str1_mxv	f90_mm_cplx8_str1_mxv
+#define F90_matmul_cplx8_str1_mxv_t	f90_mm_cplx8_str1_mxv_t
+#define F90_matmul_cplx8_str1_t		f90_mm_cplx8_str1_t
+#define F90_matmul_cplx8_str1_vxm	f90_mm_cplx8_str1_vxm
+#define F90_matmul_int1_contmxm		f90_mm_int1_contmxm
+#define F90_matmul_int1_contmxv		f90_mm_int1_contmxv
+#define F90_matmul_int1_contvxm		f90_mm_int1_contvxm
+#define F90_matmul_int1_str1		f90_mm_int1_str1
+#define F90_matmul_int1_str1_mxv	f90_mm_int1_str1_mxv
+#define F90_matmul_int1_str1_vxm	f90_mm_int1_str1_vxm
+#define F90_matmul_int2_contmxm		f90_mm_int2_contmxm
+#define F90_matmul_int2_contmxv		f90_mm_int2_contmxv
+#define F90_matmul_int2_contvxm		f90_mm_int2_contvxm
+#define F90_matmul_int2_str1		f90_mm_int2_str1
+#define F90_matmul_int2_str1_mxv	f90_mm_int2_str1_mxv
+#define F90_matmul_int2_str1_vxm	f90_mm_int2_str1_vxm
+#define F90_matmul_int4_contmxm		f90_mm_int4_contmxm
+#define F90_matmul_int4_contmxv		f90_mm_int4_contmxv
+#define F90_matmul_int4_contvxm		f90_mm_int4_contvxm
+#define F90_matmul_int4_str1		f90_mm_int4_str1
+#define F90_matmul_int4_str1_mxv	f90_mm_int4_str1_mxv
+#define F90_matmul_int4_str1_vxm	f90_mm_int4_str1_vxm
+#define F90_matmul_int8_contmxm		f90_mm_int8_contmxm
+#define F90_matmul_int8_contmxv		f90_mm_int8_contmxv
+#define F90_matmul_int8_contvxm		f90_mm_int8_contvxm
+#define F90_matmul_int8_str1		f90_mm_int8_str1
+#define F90_matmul_int8_str1_mxv	f90_mm_int8_str1_mxv
+#define F90_matmul_int8_str1_vxm	f90_mm_int8_str1_vxm
+#define F90_matmul_log1_contmxm		f90_mm_log1_contmxm
+#define F90_matmul_log1_contmxv		f90_mm_log1_contmxv
+#define F90_matmul_log1_contvxm		f90_mm_log1_contvxm
+#define F90_matmul_log2_contmxm		f90_mm_log2_contmxm
+#define F90_matmul_log2_contmxv		f90_mm_log2_contmxv
+#define F90_matmul_log2_contvxm		f90_mm_log2_contvxm
+#define F90_matmul_log4_contmxm		f90_mm_log4_contmxm
+#define F90_matmul_log4_contmxv		f90_mm_log4_contmxv
+#define F90_matmul_log4_contvxm		f90_mm_log4_contvxm
+#define F90_matmul_log8_contmxm		f90_mm_log8_contmxm
+#define F90_matmul_log8_contmxv		f90_mm_log8_contmxv
+#define F90_matmul_log8_contvxm		f90_mm_log8_contvxm
+#define F90_matmul_real16_contmxm	f90_mm_real16_contmxm
+#define F90_matmul_real16_contmxv	f90_mm_real16_contmxv
+#define F90_matmul_real16_contvxm	f90_mm_real16_contvxm
+#define F90_matmul_real4_contmxm	f90_mm_real4_contmxm
+#define F90_matmul_real4_contmxv	f90_mm_real4_contmxv
+#define F90_matmul_real4_contvxm	f90_mm_real4_contvxm
+#define F90_matmul_real4_str1		f90_mm_real4_str1
+#define F90_matmul_real4_str1_mxv	f90_mm_real4_str1_mxv
+#define F90_matmul_real4_str1_mxv_t	f90_mm_real4_str1_mxv_t
+#define F90_matmul_real4_str1_t		f90_mm_real4_str1_t
+#define F90_matmul_real4_str1_vxm	f90_mm_real4_str1_vxm
+#define F90_matmul_real8_contmxm	f90_mm_real8_contmxm
+#define F90_matmul_real8_contmxv	f90_mm_real8_contmxv
+#define F90_matmul_real8_contvxm	f90_mm_real8_contvxm
+#define F90_matmul_real8_str1		f90_mm_real8_str1
+#define _F90_matmul_real8_str1a		_f90_mm_real8_str1a
+#define _F90_matmul_real8_str1b		_f90_mm_real8_str1b
+#define _F90_matmul_real8_str1c		_f90_mm_real8_str1c
+#define F90_matmul_real8_str1_mxv	f90_mm_real8_str1_mxv
+#define F90_matmul_real8_str1_mxv_t	f90_mm_real8_str1_mxv_t
+#define F90_matmul_real8_str1_vxm	f90_mm_real8_str1_vxm
 #endif /* DESC_I8 */
